packages/relay/src/lib/eth.ts

```diff
@@ -1284,6 +1285,32 @@ export class EthImpl implements Eth {
         throw predefined.NON_EXISTING_ACCOUNT(call.from);
       }
     }
+
+    // verify gas is withing the allowed range
+    if (call.gas && call.gas > constants.BLOCK_GAS_LIMIT) {
```

Comment:
Great confirmation. Was wondering if this should be a separate config.
Although i think it should be 15 m all around though.

Reply:
@Nana-EC I don't think the mirror node should be fixed to 15m, maybe use that as a default but imo it should be configurable by the relay provider as popular providers(Alchemy and Infura) typically have a gas limit of 50m or 100m.

Reply:
I see. We're likely to both start there and build out as scenarios require.
What's the value in cases above 15m when the network limit is 15m?

Notably I did add a `CONTRACT_CALL_GAS_LIMIT` config

Reply:
@Nana-EC one usecase that I can think of is multicalls, e.g. say you want to make a 100 eth_calls where each of those calls costs ~1m gas you can instead combine all 100 into a single multicall eth_call and a `CONTRACT_CALL_GAS_LIMIT` of 100m should be sufficient.

Reply:
Understood.
Unfortunately, such a scenario would still hit the 15m gas limit per the current design.
We could revisit that limit on the mirror node later and match the relay accordingly.
For now I would advise conducting multi calls that adhere to this limit to avoid getting error responses.